File: v2ray-engine/src/main/java/libv2ray/Libv2ray.java
package libv2ray;

public class Libv2ray {
    public static void initCoreEnv(String assetPath, String deviceId) {
        // Stub: actual implementation in native AAR
    }

    public static String checkVersionX() {
        return "stub-1.0.0";
    }

    public static long measureOutboundDelay(String config, String testUrl) {
        return -1L;
    }

    public static CoreController newCoreController(CoreCallbackHandler handler) {
        return new CoreController(handler);
    }
}

File: v2ray-engine/src/main/java/libv2ray/Libv2ray.kt
package libv2ray

/** Stub until `libv2ray` AAR is placed under `v2ray-engine/libs/`. */
object Libv2ray {
  @JvmStatic
  fun initCoreEnv(assetPath: String, deviceId: String) {}

  @JvmStatic
  fun newCoreController(handler: CoreCallbackHandler): CoreController = CoreController()

  @JvmStatic
  fun checkVersionX(): String = "stub"

  @JvmStatic
  fun measureOutboundDelay(config: String, testUrl: String): Long = -1L
}
